File: smoke_tests/ruby/weak_crypto.rb
# Weak Cryptography vulnerabilities in Ruby
require 'digest'
require 'openssl'

class WeakCryptoVulnerabilities
  def hash_md5(input)
    # VULNERABLE: MD5 is cryptographically broken
    Digest::MD5.hexdigest(input)
  end

  def hash_sha1(input)
    # VULNERABLE: SHA1 is deprecated
    Digest::SHA1.hexdigest(input)
  end

  def encrypt_des(data, key)
    # VULNERABLE: DES is obsolete
    cipher = OpenSSL::Cipher.new('DES-ECB')
    cipher.encrypt
    cipher.key = key
    cipher.update(data) + cipher.final
  end

  def weak_random
    # VULNERABLE: Non-cryptographic random
    rand(1000000)
  end

  def weak_token
    # VULNERABLE: Predictable token
    Time.now.to_i.to_s
  end

  def weak_session_id
    # VULNERABLE: Predictable session
    "session_#{rand(10000)}"
  end

  def ecb_encryption(data, key)
    # VULNERABLE: ECB mode
    cipher = OpenSSL::Cipher.new('AES-128-ECB')
    cipher.encrypt
    cipher.key = key
    cipher.update(data) + cipher.final
  end
end

File: smoke_tests/php/weak_crypto.php
<?php
// Weak Cryptography vulnerabilities in PHP

class WeakCryptoVulnerabilities {
    public function hashMd5($input) {
        // VULNERABLE: MD5 is cryptographically broken
        return md5($input);
    }

    public function hashSha1($input) {
        // VULNERABLE: SHA1 is deprecated
        return sha1($input);
    }

    public function encryptDes($data, $key) {
        // VULNERABLE: DES is obsolete
        return openssl_encrypt($data, 'DES-ECB', $key);
    }

    public function generateToken() {
        // VULNERABLE: Non-cryptographic random
        return rand(0, 1000000);
    }

    public function weakSessionId() {
        // VULNERABLE: Predictable session ID
        return time();
    }

    public function ecbEncryption($data, $key) {
        // VULNERABLE: ECB mode
        return openssl_encrypt($data, 'AES-128-ECB', $key);
    }

    public function weakRandom() {
        // VULNERABLE: mt_rand is not cryptographically secure
        return mt_rand();
    }

    public function insecureHash($password) {
        // VULNERABLE: Unsalted hash
        return hash('sha256', $password);
    }
}
